src/search-scenario/azext_search_scenario/custom.py

```diff
@@ -0,0 +1,312 @@
+            else:
+                args.append('none')
+        else:
+            args.append(output_format)
+
+    if not catch_exception:
+        exit_code = ctx_cmd.cli_ctx.invoke(args)
+
+    else:
+        try:
+            exit_code = ctx_cmd.cli_ctx.invoke(args)
+        except Exception:   # pylint: disable=broad-except
+            return -1
+        except SystemExit:
+            return -1
+
+    if output_format == 'status' and exit_code == 0:
+        print()
+        from .utils import print_successful_styled_text
+        print_successful_styled_text('command completed\n')
+
+    return exit_code
+
+
+def _get_command_item_sample(command):
+    if "example" in command and command["example"]:
+        command_sample, _ = _parse_argument_value_sample(command["example"].replace(" $", " "))
+        return command_sample
+
+    from knack import help_files
+    parameter = []
+    if "arguments" in command and command["arguments"]:
+        parameter = command["arguments"]
+        sorted_param = sorted(parameter)
+        cmd_help = help_files._load_help_file(command['command'])   # pylint: disable=protected-access
+        if cmd_help and 'examples' in cmd_help and cmd_help['examples']:
+            for cmd_example in cmd_help['examples']:
+                command_sample, example_arguments = _parse_argument_value_sample(cmd_example['text'])
+                if sorted(example_arguments) == sorted_param:
+                    return command_sample
+
+    command = command["command"] if command["command"].startswith("az ") else "az " + command["command"]
+    command_sample = f"{command} {' '.join(parameter) if parameter else ''}"
+    return [(Style.PRIMARY, command_sample)]
+
+
+def _parse_argument_value_sample(command_sample):
+    if not command_sample:
+        return
+
+    cmd_items = command_sample.split()
+    arguments_start = False
+    example_arguments = []
+    command_item = []
+    argument_values = {}
+    values = []
+    for item in cmd_items:
+        if item.startswith('-'):
+            arguments_start = True
+            if values and example_arguments:
+                argument_values[example_arguments[-1]] = values
+                values = []
+            example_arguments.append(item)
+        elif not arguments_start:
+            command_item.append(item)
+        else:
+            values.append(item)
+    if values and example_arguments:
+        argument_values[example_arguments[-1]] = values
+
+    formatted_example = [(Style.PRIMARY, ' '.join(command_item))]
+    for argument in example_arguments:
+        formatted_example.append((Style.PRIMARY, " " + argument))
+        if argument in argument_values and argument_values[argument]:
+            formatted_example.append((Style.WARNING, ' <' + ' '.join(argument_values[argument]) + '>'))
+
+    return formatted_example, example_arguments
+
+
+def send_feedback(scope, option, keyword, search_results=None, adoption=None):
```

Comment:
Add some comments to help reviewers to understand

Reply:
Done